tool/tsh/proxy.go

```diff
@@ -337,6 +337,10 @@ func onProxyCommandDB(cf *CLIConf) error {
 		return trace.BadParameter("Snowflake proxy works only in the tunnel mode. Please add --tunnel flag to enable it")
 	}
 
+	if routeToDatabase.Protocol == defaults.ProtocolElasticsearch && !cf.LocalProxyTunnel {
+		return trace.BadParameter("Elasticsearch proxy works only in the tunnel mode. Please add --tunnel flag to enable it")
+	}
```

Comment:
Nit: can we extract and combine the ` routeToDatabase.Protocol == defaults.ProtocolSnowflake && !cf.LocalProxyTunnel` and `routeToDatabase.Protocol == defaults.ProtocolElasticsearch && !cf.LocalProxyTunnel` statments into separate func.

Reply:
I'm struggling to find a cleaner expression without sacrificing the readability of the code or modifying the useful error message. Since this is nit I'll go ahead with the merge with the option to revisit if I can see an acceptable option.

Reply:
I was thinking to moving the `routeToDatabase.Protocol == defaults.ProtocolSnowflake && !cf.LocalProxyTunne` and `routeToDatabase.Protocol == defaults.ProtocolElasticsearch && !cf.LocalProxyTunnel` to a separate function 
Or we can just create array `requiresLocalProxy = []string{defaults.ProtocolElasticsearch, defaults.ProtocolSnowflake)` and check  
`!cf.LocalProxyTunnel && utils.Contains(routeToDatabase.Protocol,   requiresLocalProxy) `

Reply:
Yeah, the check is just one part though: the message is the other. We want the message to vary based on the protocol in question, so either we copy it (current approach) or construct it programmatically.

- `trace.BadParameter("%v proxy works only in the tunnel mode. Please add --tunnel flag to enable it", routeToDatabase.Protocol)` gets the case wrong. 
- `trace.BadParameter("%v proxy works only in the tunnel mode. Please add --tunnel flag to enable it", strings.Title(routeToDatabase.Protocol))` is short, but strings.Title is deprecated (even though it is fine in this context).
- The replacement of `strings.Title` takes a lot more parameters, is complex and generally hard sell for what is really capitalizing of single letter.

The shortest alternative code of semi-acceptable complexity is this:

```golang
	protocols := map[string]string{
		defaults.ProtocolSnowflake:     "Snowflake",
		defaults.ProtocolElasticsearch: "Elasticsearch",
	}

	if name, ok := protocols[routeToDatabase.Protocol]; ok && !cf.LocalProxyTunnel {
		return trace.BadParameter("%v proxy works only in the tunnel mode. Please add --tunnel flag to enable it", name)
	}
```

... but I think it is harder to read than existing code.

Thoughts?

Reply:
We've added this function recently probably handy in this case
https://github.com/gravitational/teleport/blob/7744f72c6eb631791434b648ba41083b5f6d2278/lib/defaults/defaults.go#L501-L506

Reply:
This is neat, thanks! I'll fix the code in the next PR I have planned.